### DISCUSSION TITLE
yarn settings to enforce yarn and node version

### DISCUSSION THREAD

**Discussion by wdfinch** [yarnpkg/berry#3064] [Q&A] [answered]
Hi all, I was wondering if it was possible or what your thoughts were on the following. 

I would like to enforce a specific version or version range for both node and yarn as these are the two main tools used that are not managed by `package.json` (installed outside the project). In other words, if the user runs `yarn run some-command`, I would like an error to be thrown if either `node` or `yarn` does not match the version I set. Is there any way to do this? 

I know that `yarn set version` will install a bin locally in the `.yarn/releases` folder, however, I don't want to commit a binary to version control. The `yarnPath` settings could sort of handle this use case as it would point to bin with a certain version name but this can be easily overwritten and doesn't seem like the right solution.

**Comment by arcanis** [comment, marked as answer]
For locking Yarn versions, [Corepack](https://github.com/nodejs/corepack) will be the way forward. In the meantime (and taking into account that it'll take a while for Corepack to settle in the ecosystem), checking-in is the suggested approach.

**Reply by wdfinch** [reply]
Thanks!
